app/Providers/Filament/AdminPanelProvider.php

```diff
@@ -179,10 +177,26 @@ public function panel(Panel $panel): Panel
                     ->url(fn () => EditProfile::getUrl())
                     ->icon('heroicon-s-cog-6-tooth'),
             ])
+            ->colors(fn (ThemeSettings $themeSettings): array => array_merge(config('default-colors'), Feature::active('synced_theme_settings') ? $themeSettings->color_overrides : []))
             ->renderHook(
                 'panels::scripts.before',
                 fn () => view('filament.scripts.scroll-sidebar-to-active-menu-item'),
-            );
+            )
+            ->renderHook(
+                'panels::head.end',
+                fn (ThemeSettings $themeSettings) => (Feature::active('synced_theme_settings') && $themeSettings->url) ? view('filament.layout.theme', ['url' => $themeSettings->url]) : null,
+            )
+            ->bootUsing(function (Panel $panel) {
+                if (! Tenant::current()) {
+                    return;
+                }
+
+                if (! Feature::active('synced_theme_settings')) {
+                    return;
+                }
+
+                $panel->darkMode(app(ThemeSettings::class)->has_dark_mode);
+            });
```

Comment:
With us still referencing the settings class here does this mean we still can only run composer commands in the container?

Reply:
I don't believe so. All references are within functions. The colors function only needs to be called on render to output the layout. The render hooks function only gets called on render. The boot function only runs in middleware on pages within that panel.